modules/sdk/src/main/scala/org/tessellation/sdk/domain/healthcheck/consensus/HealthCheckConsensus.scala

```diff
@@ -29,25 +29,28 @@ abstract class HealthCheckConsensus[
   F[_]: Async: GenUUID,
   K <: HealthCheckKey,
   A <: HealthCheckStatus,
-  B <: ConsensusHealthStatus[K, A]: TypeTag
+  B <: ConsensusHealthStatus[K, A]: TypeTag,
+  C <: HealthCheckConsensusDecision
 ](
   clusterStorage: ClusterStorage[F],
   selfId: PeerId,
-  driver: HealthCheckConsensusDriver[K, A, B],
+  driver: HealthCheckConsensusDriver[K, A, B, C],
   gossip: Gossip[F],
   config: HealthCheckConfig
 ) extends HealthCheck[F] {
   def logger = Slf4jLogger.getLogger[F]
 
-  def allRounds: Ref[F, ConsensusRounds[F, K, A, B]]
+  def allRounds: Ref[F, ConsensusRounds[F, K, A, B, C]]
 
   def ownStatus(key: K): F[Fiber[F, Throwable, A]]
 
   def statusOnError(key: K): A
 
   def periodic: F[Unit]
 
-  def roundsInProgress: F[InProgress[F, K, A, B]] = allRounds.get.map(_.inProgress)
+  def onOutcome(peers: ConsensusRounds.Outcome[F, K, A, B, C]): F[Unit]
```

Comment:
Shouldn't it be called `outcomes` instead of `peers`?

Reply:
I just updated the type by additional type parameter

Reply:
:+1: fixed